src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CipherData.cs

```diff
@@ -63,7 +63,7 @@ public byte[] CipherValue
 
         public XmlElement GetXml()
         {
-            if (CacheValid) return _cachedXml;
+            if (CacheValid) return _cachedXml!;
```

Comment:
Mark `CacheValid` with `[MemberNotNullWhen(true, nameof(_cachedXml))]`

Reply:
Ah, yes, that's the attribute I was looking for. I was aware that there was a `NotNullWhen...` attribute.  I'll do this shortly, thanks for the info!

Reply:
Great! I've updated this (and other types that have this pattern). One thing though: should these attributes also end up in the `ref` assembly? When I run `dotnet msbuild /t:GenerateReferenceAssemblySource`, it regenerates the file, but further builds result in compilation errors.

Reply:
Yes, attributes and nullable annotations should be reflected in `ref`. 
I'd been adding all of those manually, but in theory nullable annotations should be updated in `ref` after running `dotnet msbuild /t:GenerateReferenceAssemblySource`